src/java.security.jgss/share/native/libj2gss/NativeUtil.c

```diff
@@ -463,31 +471,33 @@ void throwOutOfMemoryError(JNIEnv *env, const char *message) {
 jstring getJavaString(JNIEnv *env, gss_buffer_t bytes) {
   jstring result = NULL;
   OM_uint32 minor;
-  int len;
+  jsize len;
   jbyteArray jbytes;
 
-  if (bytes != NULL) {
-    /* constructs the String object with new String(byte[])
-       NOTE: do NOT include the trailing NULL */
-    len = (int) bytes->length;
-    jbytes = (*env)->NewByteArray(env, len);
-    if (jbytes == NULL) {
-      goto finish;
-    }
+  if (bytes == NULL) {
+    return NULL;
+  }
 
-    (*env)->SetByteArrayRegion(env, jbytes, 0, len, (jbyte *) bytes->value);
-    if ((*env)->ExceptionCheck(env)) {
-      goto finish;
-    }
+  /* constructs the String object with new String(byte[]) */
```

Comment:
This was pre-existing, but it's likely unsafe. This will use whatever the default Java character encoding is. You probably want to use an explicit code page.

Reply:
And if you're not going to do that, it would be simpler to decode the name yourself and use `(*env)->NewString(...)`